tsdb/engine/tsm1/file_store.go

```diff
@@ -782,12 +840,25 @@ func newKeyCursor(fs *FileStore, key string, t int64, ascending bool) *KeyCursor
 		sort.Sort(descLocations(c.seeks))
 	}
 
+	// Determine the distinct set of TSM files in use and mark then as in-use
+	for _, f := range c.seeks {
+		if _, ok := c.refs[f.r.Path()]; !ok {
+			f.r.Ref()
```

Comment:
I think this is ok, but I want to highlight it just to make sure I'm reading the code correctly. I think that both the `Replace` and `newKeyCursor` use the same lock so they can't be run simultaneously. Without the lock, there's a possibility of a race condition here where `fs.location()` retrieves a `TSMFile` and then that file gets deleted before the reference can be taken. I think because of the lock, this is fine, but I just want to confirm that's correct because this would be a very hard race condition to find otherwise.